pkg/sql/backfill.go

```diff
@@ -677,17 +681,46 @@ func runSchemaChangesInTxn(
 func columnBackfillInTxn(
 	ctx context.Context,
 	txn *client.Txn,
+	tc *TableCollection,
 	evalCtx *tree.EvalContext,
 	tableDesc *sqlbase.TableDescriptor,
 ) error {
+	// A column backfill in the ADD state is a noop.
```

Comment:
Are you sure?

What about:

```sql
BEGIN;
CREATE TABLE blah ...;
INSERT INTO blah VALUES ...;
CREATE INDEX blih ON blah (...);  -- this needs a backfill!
COMMIT;
```

Reply:
If TABLE blah is in the ADD state, the INSERT will fail

Reply:
oh I see thanks!